src/index.ts

```diff
@@ -30,29 +30,58 @@ export function validationMetadatasToSchemas(userOptions?: Partial<IOptions>) {
       const target = ownMetas[0].target as Function
       const metas = ownMetas.concat(getInheritedMetadatas(target, metadatas))
 
-      const properties = _(metas)
-        .groupBy('propertyName')
-        .mapValues((propMetas, propKey) => {
-          const schema = applyConverters(propMetas, options)
-          return applyDecorators(schema, target, options, propKey)
-        })
-        .value()
-
-      const definitionSchema: SchemaObject = {
-        properties,
-        type: 'object',
-      }
+      return buildJsonSchemaForSpecificTarget(metadatas, target, metas, options)
+    })
+    .value()
 
-      const required = getRequiredPropNames(target, metas, options)
-      if (required.length > 0) {
-        definitionSchema.required = required
-      }
+  return schemas
+}
 
-      return applyDecorators(definitionSchema, target, options, target.name)
+/**
+ * Convert class-validator class into JSON Schema definition.
+ */
+export type ClassType<T> = new (...args: any[]) => T;
```

Comment:
```suggestion
export type ClassType = new (...args: any[]) => any;
```

We can omit the unused generic here and below I think.